src/List/ListItemAvatar.js

```diff
@@ -16,10 +16,21 @@ export const styleSheet = createStyleSheet('MuiListItemAvatar', () => {
   };
 });
 
+/**
+ * `<ListItemAvatar>` is a simple wrapper to apply the `dense` mode styles to `Avatar`.
+ *
+ * ```
+ * <ListItemAvatar>
+ *   <Avatar>
+ * </ListItemAvatar>
+ * ```
+ */
 export default function ListItemAvatar(props, context) {
   if (!context.dense) {
-    warning(false, `Material-UI: <ListItemAvatar/> is a simple wrapper for the dense mode.
-      You do not need it.`);
+    warning(false, `Material-UI: <ListItemAvatar> is a simple wrapper to apply the dense styles 
+      to Avatar. You do not need it.`);
+    // TODO: `dense` might be set according to platform the app is running on.
```

Comment:
It depends on the vision, the warning is here to save users rendering a useless `ListItemAvatar` component when dense is `false`. We could do the opposite for consistency.
What do you have in mind by according to the platform? Isn't something depending on the context, not the platform?